build/Dockerfile.build

```diff
@@ -0,0 +1,35 @@
+
+# The docker image for the actual build process.
+# Copy in only the necessary files for building.
+FROM ghcr.io/istio-ecosystem/authservice/authservice-build-env:2021-11-26 as auth-builder
+COPY . /src/
+
+# Build auth binary.
+ARG bazel_flags=""
+WORKDIR /src
+RUN bash ./bazel/setup_clang.sh /usr/lib/llvm-12/
+RUN BAZEL_FLAGS=$bazel_flags make bazel-bin/src/main/auth_server
+
+# Create our final auth-server container image.
+FROM debian:buster
+RUN groupadd -r auth-server-grp && useradd -m -g auth-server-grp auth-server-usr
+
+# Install dependencies
+RUN apt update && apt upgrade -y && apt install -y --no-install-recommends \
+    ca-certificates  \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=auth-builder \
+     /src/bazel-bin/src/main/auth_server \
+     /src/bazel-bin/external/boost/libboost_chrono.so.1.70.0 \
+     /src/bazel-bin/external/boost/libboost_context.so.1.70.0 \
+     /src/bazel-bin/external/boost/libboost_coroutine.so.1.70.0 \
+     /src/bazel-bin/external/boost/libboost_thread.so.1.70.0 \
+     /app/
+
+ENV LD_LIBRARY_PATH=.
+RUN chgrp auth-server-grp /app/* && chown auth-server-usr /app/* && chmod u+x /app/*
+
+USER auth-server-usr
+WORKDIR /app
+ENTRYPOINT ["/app/auth_server"]
```

Comment:
NOTE(incfly): TEST BEFORE MERGE.

Test the actual e2e deployment again to ensure things work e2e.